FWCore/ParameterSet/python/SequenceTypes.py

```diff
@@ -1537,6 +1537,16 @@ def copyAndExclude(self,listOfModulesToExclude):
         v = _CopyAndExcludeSequenceVisitor(listOfModulesToExclude)
         self.visit(v)
         return Task(*v.result(self))
+    def copyAndAdd(self, *modulesToAdd):
+        """Returns a copy of the Task adding those modules/tasks in 'modulesToAdd'.
+        The 'modulesToAdd' can be passed in either as a single list, or as many arguments directly.
+        """
+        t = self.copy()
+        if len(modulesToAdd) == 1 and not isinstance(modulesToAdd[0], _ConfigureComponent):
+            t.add(*modulesToAdd[0])
+        else:
+            t.add(*modulesToAdd)
```

Comment:
@Dr15Jones The `copyAndExclude()` takes a list of modules
```py
task.copyAndExclude([a, b])
```
and for consistency I wanted this function to have similar interface. On the other hand, I'd like more of passing the modules directly along
```py
task.copyAndAdd(a, b)
```
so I included support for both styles. What do you think?

(if ok, it does raise the question on the interface of `copyAndExclude()` )

Reply:
So a python list has two ways to _add_, one is `append` which takes a single element argument and the other is `extend` which takes an _iterable_. `Task` is actually closer the concept of a `set` and a `set` has `add` which adds a single item and `update` which also takes _iterable_.

So to be somewhat consistent with python containers, I'd go with passing _iterable_ rather than elements individually.

Reply:
Thanks, consistency with python list/set makes sense in general. On the other hand, `Task.add()` already takes elements individually
https://github.com/cms-sw/cmssw/blob/577520f038d7c6bbd3afc0d9d3242fe084db1750/FWCore/ParameterSet/python/SequenceTypes.py#L1432-L1438